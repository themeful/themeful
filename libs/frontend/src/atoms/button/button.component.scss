:host {
  display: block;
}

.button {
  border-radius: 0.25rem;
  border: none;
  padding: 8px 16px;
  min-width: 20px;
  min-height: 20px;
  font-weight: 600;
  display: inline-flex;
  cursor: pointer;
  align-items: center;
  justify-content: center;

  --bg-opacity: 0.5;
  --ggs: 0.5;

  &.active {
    --bg-opacity: 0.75;
  }

  &.primary {
    background-color: rgba(50, 90, 240, var(--bg-opacity));
  }

  &.secondary {
    background-color: rgba(200, 200, 200, var(--bg-opacity));
  }

  &.danger {
    background-color: rgba(255, 100, 100, var(--bg-opacity));
  }

  &:disabled {
    filter: grayscale(100%);
    cursor: default;
  }

  &:focus {
    outline: 0 !important;
    box-shadow: 0 0 0 2px rgba(0, 0, 0, 0.5);
  }

  &:focus:not(:focus-visible) {
    box-shadow: none !important;
  }

  &:hover:not(.active):not(:disabled) {
    filter: brightness(0.75);
  }
}

//   &--icon {
//     padding: 2px;
//     width: 24px;
//     height: 24px;
//   }

//   &--small {
//     padding: 4px 8px;
//     font-weight: 400;
//   }

//   & + .m4-button {
//     margin-left: 4px;
//   }

//   &-group {
//     display: flex;

//     & > * {
//       margin-left: 0 !important;
//       flex-grow: 4;

//       &:not(:first-child) {
//         border-top-left-radius: 0;
//         border-bottom-left-radius: 0;
//       }
//       &:not(:last-child) {
//         border-top-right-radius: 0;
//         border-bottom-right-radius: 0;
//       }
//     }
//   }
// }
