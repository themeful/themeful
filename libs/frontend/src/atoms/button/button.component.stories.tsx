import { h } from '@stencil/core'

export default {
  title: 'Atoms/Buttons',
  args: {
    buttons: [
      { label: 'Primary', args: { type: 'button', kind: 'primary' } },
      { label: 'Secondary', args: { type: 'submit', kind: 'secondary' } },
      { label: 'Danger', args: { type: 'button', kind: 'danger' } },
    ],
  },
}

export const buttons = ({ buttons }): HTMLElement => {
  return (
    <div>
      <h1>Buttons</h1>
      <table style={{ borderSpacing: '1rem' }}>
        <tr>
          <td>
            <h3>default</h3>
          </td>
          <td>
            <h3>active</h3>
          </td>
          <td>
            <h3>disabled</h3>
          </td>
        </tr>
        {buttons.map((button) => {
          return (
            <tr>
              <td>
                <tf-button {...button.args}>{button.label}</tf-button>
              </td>
              <td>
                <tf-button {...button.args} active>
                  {button.label}
                </tf-button>
              </td>
              <td>
                <tf-button {...button.args} disabled>
                  {button.label}
                </tf-button>
              </td>
            </tr>
          )
        })}
      </table>
      <h1>With Icon</h1>
      <table>
        <tr>
          <td>default</td>
          <td>
            <tf-button kind="secondary">
              <tf-icon icon="sync"></tf-icon>Some Text
            </tf-button>
          </td>
          <td>
            <tf-button kind="secondary">
              Some Text<tf-icon icon="sync"></tf-icon>
            </tf-button>
          </td>
          <td>
            <tf-button kind="secondary">
              <tf-icon icon="sync"></tf-icon>
            </tf-button>
          </td>
        </tr>
        <tr>
          <td>small</td>
          <td>
            <tf-button kind="secondary" size="small">
              <tf-icon icon="sync"></tf-icon>Some Text
            </tf-button>
          </td>
          <td>
            <tf-button kind="secondary" size="small">
              Some Text<tf-icon icon="sync"></tf-icon>
            </tf-button>
          </td>
          <td>
            <tf-button kind="secondary" size="small">
              <tf-icon icon="sync"></tf-icon>
            </tf-button>
          </td>
        </tr>
        <tr>
          <td>icon</td>
          <td>
            <tf-button kind="secondary" size="icon">
              <tf-icon icon="sync"></tf-icon>Some Text
            </tf-button>
          </td>
          <td>
            <tf-button kind="secondary" size="icon">
              Some Text<tf-icon icon="sync"></tf-icon>
            </tf-button>
          </td>
          <td>
            <tf-button kind="secondary" size="icon">
              <tf-icon icon="sync"></tf-icon>
            </tf-button>
          </td>
        </tr>
      </table>
    </div>
  )
}
