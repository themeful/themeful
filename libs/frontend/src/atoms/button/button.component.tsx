import { Component, Event, h, EventEmitter, Prop } from '@stencil/core'
import { ButtonKind } from './interfaces'

@Component({
  tag: 'tf-button',
  styleUrl: 'button.component.scss',
})
export class TfButtonComponent {
  /** Button type*/
  @Prop() type = 'button'

  /** Button kind */
  @Prop() kind: ButtonKind = 'primary'

  /** Active state */
  @Prop() active: false

  /** Disabled state */
  @Prop() disabled: false

  /** Event emitted when the item is clicked */
  @Event({ bubbles: true, composed: true }) clicked: EventEmitter

  render() {
    return (
      <button
        type={this.type}
        class={`button ${this.kind}${this.active ? ' active' : ''}`}
        disabled={this.disabled}
        onClick={() => {
          this.clicked.emit()
        }}
      >
        <slot />
      </button>
    )
  }
}
