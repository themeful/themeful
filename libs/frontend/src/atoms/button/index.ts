export * from './button.component'
export * from './interfaces'
