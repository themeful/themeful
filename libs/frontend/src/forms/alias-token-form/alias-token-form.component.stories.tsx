import { h } from '@stencil/core'

export default {
  title: 'Forms/Alias Token',
  args: {
    first: 'John',
    middle: 'S',
    last: 'Doe',
  },
}

export const aliasToken = (args) => {
  return (
    <div>
      <div class="header">
        <h1>Alias Token Form</h1>
      </div>
      <div class="content">
        <div style={{ position: 'relative', width: '600px' }}>
          <tf-alias-token-form {...args}></tf-alias-token-form>
        </div>
      </div>
    </div>
  )
}
