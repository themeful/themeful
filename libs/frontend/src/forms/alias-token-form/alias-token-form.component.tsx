import { Component, h, Prop } from '@stencil/core'

@Component({
  tag: 'tf-alias-token-form',
  styleUrl: 'alias-token-form.component.scss',
  shadow: true,
})
export class AliasTokenFormComponent {
  /**
   * The first name
   */
  @Prop() first: string

  /**
   * The middle name
   */
  @Prop() middle: string

  /**
   * The last name
   */
  @Prop() last: string

  private getText(): string {
    return `${this.first} ${this.middle} ${this.last}`
  }

  public render(): HTMLTfAliasTokenFormElement {
    return <div>Hello, World! {this.getText()}</div>
  }
}
