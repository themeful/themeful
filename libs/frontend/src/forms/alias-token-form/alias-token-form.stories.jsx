export default {
  title: 'Forms/Alias Token Select',
  args: {
    first: 'John',
    middle: 'S',
    last: 'Doe',
  },
}

export const aliasTokenSelect = (args) => {
  return <alias-token-form {...args}></alias-token-form>
}
