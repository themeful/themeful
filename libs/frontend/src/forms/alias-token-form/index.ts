export * from './alias-token-form.component'
