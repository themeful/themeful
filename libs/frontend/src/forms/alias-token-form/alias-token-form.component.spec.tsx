import { h } from '@stencil/core'
import { newSpecPage } from '@stencil/core/testing'
import { ButtonComponent } from '../../components/button/button.component'
import { MultiSelectInputComponent } from '../../components/inputs/multi-select-input/multi-select-input.component'
import { AliasTokenFormComponent } from './alias-token-form.component'

describe('alias-token-form', () => {
  it('renders', async () => {
    const data = {
      formData: {
        identifier: 'dtActionBg',
        aliasTokens: ['atCardBackground', 'atButtonBackground'],
        fields: {},
      },
    }
    const page = await newSpecPage({
      components: [AliasTokenFormComponent, ButtonComponent, MultiSelectInputComponent],
      autoApplyChanges: true,
      template: () => <tf-alias-token-select {...data} />,
    })

    expect(page.root).toMatchSnapshot()
  })

  it('renders with values', async () => {
    const data = {
      formData: {
        identifier: 'dtActionBg',
      },
    }
    const page = await newSpecPage({
      components: [AliasTokenFormComponent, ButtonComponent, MultiSelectInputComponent],
      autoApplyChanges: true,
      template: () => <tf-alias-token-select {...data} />,
    })

    expect(page.root).toMatchSnapshot()
  })
})
