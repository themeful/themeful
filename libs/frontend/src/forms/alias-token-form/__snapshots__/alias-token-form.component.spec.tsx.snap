// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alias-token-form renders 1`] = `<tf-alias-token-select />`;

exports[`alias-token-form renders with values 1`] = `<tf-alias-token-select />`;
